react/week4/git_user_search/src/index.js

```diff
@@ -2,71 +2,106 @@ import React from 'react';
 import ReactDOM from 'react-dom';
 import './index.css';
 import ListOfUsers from './ListOfUsers.js'
+import Header from './header.js'
 
 
 
 
 class Page extends React.Component {
     constructor(props) {
         super(props);
-        // var timeoutHandle = false;
+        this.state = { total_count: 0, items: [] }
     }
 
-    triggerTimer = () => {
+    triggerTimer = (textChangeEvent) => {
+        const searchWord = textChangeEvent.target.value.substr(0, 100)
         if (this.timeoutHandle) window.clearTimeout(this.timeoutHandle);
-        this.timeoutHandle = window.setTimeout(this.getData, 1000);
+        this.timeoutHandle = window.setTimeout(e => this.getData(searchWord), 2000);
     }
 
-    createList = (dataFromAPI) => {
-        console.log('wdfsad')
-        return (true)
-        // const listOfUsers = dataFromAPI.items.map(user => user.login)
-        // console.log(listOfUsers);
-    }
-
-    printData = e => {
-        console.log(e)
-        return (true)
-    }
-
-    createListrrr = (dataFromAPI) => {
-        console.log('wdfsad')
-        // return (true)
-        // const listOfUsers = dataFromAPI.items.map(user => user.login)
-        // console.log(listOfUsers);
-    }
-
-    getData = () => {
-        fetch('https://api.github.com/search/users?q=hosseinTa').then(
-            function (response) {
-                if (response.status > 299 || response.status < 100) {
-                    console.log('Please check your internet connection');
-                    return;
-                }
-                response.json().then(res => printData('tdffffffffffffffft'));
+    getData = async (searchWord) => {
+        fetch(`https://api.github.com/search/users?q=${searchWord}`).then(async response => {
+            if (response.status > 299 || response.status < 100) {
+                console.log('Please check your internet connection');
+                return;
             }
-        )
+            const extractData = await response.json()
+            this.setState({ ...extractData })
+        })
     }
 
-    
+
 
     render = () => {
         return (
             <div>
-                <h1>Hello World</h1>
-                <input type="text" onChange={this.triggerTimer}></input>
-                {/* <ListOfUsers listOfUsers={}></ListOfUsers> */}
+                <Header />
+                <input type="text" onChange={e => this.triggerTimer(e)}></input> {`${new Date().getHours()}:${new Date().getMinutes()}:${new Date().getSeconds()}`}
```

Comment:
This is too complicated. What does it do? You should always use a library to handle time and dates